src/nvim/memline.c

```diff
@@ -2492,6 +2492,11 @@ int ml_replace(linenr_T lnum, char *line, bool copy)
   return ml_replace_buf(curbuf, lnum, line, copy, false);
 }
 
+int ml_replace_len(linenr_T lnum, char *line, size_t len, bool copy)
+{
+  return ml_replace_buf_len(curbuf, lnum, line, len, copy, false);
```

Comment:
`len` isn't suppose to account for `NUL`. `ml_replace_buf_len()` assumes `line[len] == NUL` and `xmemdupz(line, len_arg)` follows that assumption.

Reply:
Documented